tesseract/tesseract_collision/src/fcl/fcl_discrete_managers.cpp

```diff
@@ -179,43 +255,119 @@ void FCLDiscreteBVHManager::setActiveCollisionObjects(const std::vector<std::str
   active_ = names;
 
   for (auto& co : link2cow_)
-  {
-    COW::Ptr& cow = co.second;
-
-    updateCollisionObjectFilters(active_, *cow);
-  }
+    updateCollisionObjectFilters(active_, co.second, static_manager_, dynamic_manager_);
 }
 
 const std::vector<std::string>& FCLDiscreteBVHManager::getActiveCollisionObjects() const { return active_; }
 void FCLDiscreteBVHManager::setContactDistanceThreshold(double contact_distance)
 {
   contact_distance_ = contact_distance;
+  static_update_.clear();
+  dynamic_update_.clear();
+
+  for (auto& cow : link2cow_)
+  {
+    cow.second->setContactDistanceThreshold(contact_distance_ / 2.0);
+    std::vector<CollisionObjectRawPtr>& co = cow.second->getCollisionObjectsRaw();
+    if (cow.second->m_collisionFilterGroup == CollisionFilterGroups::StaticFilter)
+    {
+      static_update_.insert(static_update_.end(), co.begin(), co.end());
+    }
+    else
+    {
+      dynamic_update_.insert(dynamic_update_.end(), co.begin(), co.end());
+    }
+  }
+
+  if (!static_update_.empty())
+    static_manager_->update(static_update_);
+
+  if (!dynamic_update_.empty())
+    dynamic_manager_->update(dynamic_update_);
 }
 
 double FCLDiscreteBVHManager::getContactDistanceThreshold() const { return contact_distance_; }
 void FCLDiscreteBVHManager::setIsContactAllowedFn(IsContactAllowedFn fn) { fn_ = fn; }
 IsContactAllowedFn FCLDiscreteBVHManager::getIsContactAllowedFn() const { return fn_; }
-void FCLDiscreteBVHManager::contactTest(ContactResultMap& collisions, const ContactTestType& type)
+
+/**
+ * @brief This is used to perform self check for fcl. The AABB Tree self check is N^2 which is slow and it is faster
+ *        to loop over the shapes and check bounding boxes.
```

Comment:
This was for testing if the self collide is slower than just looping over the collision shapes because the documentation said it is an N^2 problem. It is currently not being used but I need to do some testing to see if this is faster or slower for self collision checks using FCL.